crates/net/discv5/src/lib.rs

```diff
@@ -541,18 +541,14 @@ pub fn get_lookup_target(
     let mut target = local_node_id.raw();
 
     // make sure target has a 'log2distance'-long suffix that differs from local node id
-    if kbucket_index != 0 {
-        let suffix_bit_offset = MAX_KBUCKET_INDEX.saturating_sub(kbucket_index);
-        let suffix_byte_offset = suffix_bit_offset / 8;
-        // todo: flip the precise bit
-        // let rel_suffix_bit_offset = suffix_bit_offset % 8;
-        target[suffix_byte_offset] = !target[suffix_byte_offset];
-
-        if suffix_byte_offset != 31 {
-            for b in target.iter_mut().take(31).skip(suffix_byte_offset + 1) {
-                *b = rand::random::<u8>();
-            }
-        }
+    let suffix_bit_offset = MAX_KBUCKET_INDEX.saturating_sub(kbucket_index);
+    let suffix_byte_offset = suffix_bit_offset / 8;
+    // todo: flip the precise bit
+    // let rel_suffix_bit_offset = suffix_bit_offset % 8;
+    target[suffix_byte_offset] = !target[suffix_byte_offset];
```

Comment:
Closes https://github.com/paradigmxyz/reth/issues/7763

```suggestion
    target[suffix_byte_offset] ^= 1 << (suffix_bit_offset % 8);
```

Reply:
this flips other bits too

Reply:
here the issue says: should flip *only* the specific bit @DaniPopes

Reply:
reverting these suggestions, out of scope